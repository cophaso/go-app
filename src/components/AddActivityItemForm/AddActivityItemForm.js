import React, { Component } from 'react'
import { Button, Input, Required } from '../Utils/Utils'
import ItinerariesApiService from '../../services/Itinerary-api-service';
import ItineraryContext from '../../contexts/ItineraryContext';


export default class AddItineraryForm extends Component {
  static defaultProps = {
    onAddActivityItemSuccess: () => {}
  }

  state = {
    error: null
  };

  static contextType = ItineraryContext

  handleSubmit = ev => {
    ev.preventDefault()
    const { itineraryId } = window.location.pathname.split('/')[2]
    const { travel_type, 
            title,
            description,
            start_date, 
            end_date,
            start_time,
            end_time,
            cost,
            url
          } = ev.target

    this.setState({ error: null })

    ItinerariesApiService.postActivityItem(itineraryId, {
      travel_type: travel_type.value,
      title: title.value,
      description: description.value,
      start_date: start_date.value,
      end_date: end_date.value,
      start_time: start_time.value,
      end_time: end_time.value,
      cost: cost.value,
      url: url.value,
      itinerary_id: itineraryId,
      user_id: localStorage.getItem('user_id')
    }).then(this.context.addActivityItem)
    .then(() => {
        this.props.onAddActivityItemSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <form
      className='AddActivityItemForm'
        onSubmit={this.handleSubmit}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='title'>
          <label htmlFor='AddActivityItemForm__title'>
            Title <Required />
          </label>
          <Input
            name='title'
            type='text'
            required
            id='AddActivityItemForm__title'>
          </Input>
        </div>
        <div className='travel_type'>
          <label htmlFor='AddActivityItemForm__travel-type'>
            Travel Type 
          </label>
          <select name="travel_type">
            <option value="Accomodations">Accomodations</option>
            <option value="Rentals">Rentals</option>
            <option value="Activity">Activity</option>
            <option value="Event">Event</option>
            <option value="Dining and Bars">Dining and Bars</option>
          </select>
        </div>
        <div className='description'>
          <label htmlFor='AddActivityItemForm__description'>
            Description 
          </label>
          <Input
            name='description'
            type='textarea'
            id='AddActivityItemForm__title'>
          </Input>
        </div>
        <div className='start_date'>
          <label htmlFor='AddActivityItemForm__start_date'>
            Start Date
          </label>
          <Input
            name='start_date'
            type='date' 
            id='AddActivityItemForm__start_date'>
          </Input>
        </div>
        <div className='end_date'>
          <label htmlFor='AddActivityItemForm__end_date'>
            End Date
          </label>
          <Input
            name='end_date'
            type='date' 
            id='AddActivityItemForm__end_date'>
          </Input>
        </div>
        <div className='start_time'>
          <label htmlFor='AddActivityItemForm__start_time'>
            Start Time
          </label>
          <Input
            name='start_time'
            type='time' 
            id='AddActivityItemForm__start_time'>
          </Input>
        </div>
        <div className='end_time'>
          <label htmlFor='AddActivityItemForm__end_time'>
            End Time
          </label>
          <Input
            name='end_time'
            type='time' 
            id='AddActivityItemForm__end_time'>
          </Input>
        </div>
        <div className='cost'>
          <label htmlFor='AddActivityItemForm__cost'>
            Cost
          </label>
          <Input
            name='cost'
            type='number' 
            step='any'
            id='AddActivityItemForm__cost'>
          </Input>
        </div>
        <div className='url'>
          <label htmlFor='AddActivityItemForm__url'>
            URL
          </label>
          <Input
            name='url'
            type='text' 
            id='AddActivityItemForm__url'>
          </Input>
        </div>
        <Button type='submit'>
          Add
        </Button>
      </form>
    )}
}