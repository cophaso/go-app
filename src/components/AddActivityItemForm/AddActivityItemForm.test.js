import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddActivityItemForm from './AddActivityItemForm';

describe(`AddActivityItemForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a form.AddActivityItemForm by default', () => {
    const wrapper = shallow(<AddActivityItemForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the AddActivityItemForm given props', () => {
    const wrapper = shallow(<AddActivityItemForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
