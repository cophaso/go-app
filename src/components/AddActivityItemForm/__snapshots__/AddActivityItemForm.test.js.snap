// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivityItemForm component renders a form.AddActivityItemForm by default 1`] = `
<form
  className="AddActivityItemForm"
  onSubmit={[Function]}
>
  <div
    className="title"
  >
    <label
      htmlFor="AddActivityItemForm__title"
    >
      Title 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="travel_type"
  >
    <label
      htmlFor="AddActivityItemForm__travel-type"
    >
      Travel Type
    </label>
    <select
      name="travel_type"
    >
      <option
        value="Accommodations"
      >
        Accommodations
      </option>
      <option
        value="Rentals"
      >
        Rentals
      </option>
      <option
        value="Activity"
      >
        Activity
      </option>
      <option
        value="Event"
      >
        Event
      </option>
      <option
        value="Dining and Bars"
      >
        Dining and Bars
      </option>
    </select>
  </div>
  <div
    className="description"
  >
    <label
      htmlFor="AddActivityItemForm__description"
    >
      Description
    </label>
    <Input
      id="AddActivityItemForm__title"
      name="description"
      type="textarea"
    />
  </div>
  <div
    className="start_date"
  >
    <label
      htmlFor="AddActivityItemForm__start_date"
    >
      Start Date 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__start_date"
      name="start_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="end_date"
  >
    <label
      htmlFor="AddActivityItemForm__end_date"
    >
      End Date 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__end_date"
      name="end_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="start_time"
  >
    <label
      htmlFor="AddActivityItemForm__start_time"
    >
      Start Time 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__start_time"
      name="start_time"
      required={true}
      type="time"
    />
  </div>
  <div
    className="end_time"
  >
    <label
      htmlFor="AddActivityItemForm__end_time"
    >
      End Time 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__end_time"
      name="end_time"
      required={true}
      type="time"
    />
  </div>
  <div
    className="cost"
  >
    <label
      htmlFor="AddActivityItemForm__cost"
    >
      Cost 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__cost"
      name="cost"
      required={true}
      step="any"
      type="number"
    />
  </div>
  <div
    className="url"
  >
    <label
      htmlFor="AddActivityItemForm__url"
    >
      URL
    </label>
    <Input
      id="AddActivityItemForm__url"
      name="url"
      type="text"
    />
  </div>
  <div
    role="alert"
  />
  <Button
    type="submit"
  >
    Add
  </Button>
</form>
`;

exports[`AddActivityItemForm component renders the AddActivityItemForm given props 1`] = `
<form
  className="AddActivityItemForm"
  onSubmit={[Function]}
>
  <div
    className="title"
  >
    <label
      htmlFor="AddActivityItemForm__title"
    >
      Title 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="travel_type"
  >
    <label
      htmlFor="AddActivityItemForm__travel-type"
    >
      Travel Type
    </label>
    <select
      name="travel_type"
    >
      <option
        value="Accommodations"
      >
        Accommodations
      </option>
      <option
        value="Rentals"
      >
        Rentals
      </option>
      <option
        value="Activity"
      >
        Activity
      </option>
      <option
        value="Event"
      >
        Event
      </option>
      <option
        value="Dining and Bars"
      >
        Dining and Bars
      </option>
    </select>
  </div>
  <div
    className="description"
  >
    <label
      htmlFor="AddActivityItemForm__description"
    >
      Description
    </label>
    <Input
      id="AddActivityItemForm__title"
      name="description"
      type="textarea"
    />
  </div>
  <div
    className="start_date"
  >
    <label
      htmlFor="AddActivityItemForm__start_date"
    >
      Start Date 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__start_date"
      name="start_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="end_date"
  >
    <label
      htmlFor="AddActivityItemForm__end_date"
    >
      End Date 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__end_date"
      name="end_date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="start_time"
  >
    <label
      htmlFor="AddActivityItemForm__start_time"
    >
      Start Time 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__start_time"
      name="start_time"
      required={true}
      type="time"
    />
  </div>
  <div
    className="end_time"
  >
    <label
      htmlFor="AddActivityItemForm__end_time"
    >
      End Time 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__end_time"
      name="end_time"
      required={true}
      type="time"
    />
  </div>
  <div
    className="cost"
  >
    <label
      htmlFor="AddActivityItemForm__cost"
    >
      Cost 
      <Required />
    </label>
    <Input
      id="AddActivityItemForm__cost"
      name="cost"
      required={true}
      step="any"
      type="number"
    />
  </div>
  <div
    className="url"
  >
    <label
      htmlFor="AddActivityItemForm__url"
    >
      URL
    </label>
    <Input
      id="AddActivityItemForm__url"
      name="url"
      type="text"
    />
  </div>
  <div
    role="alert"
  />
  <Button
    type="submit"
  >
    Add
  </Button>
</form>
`;
